stablehlo/tests/interpret/dynamic_broadcast_in_dim.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+// %operand: [
+//            [1, 2, 3]
+//           ]
+func.func @broadcast_in_dim() {
+  %operand = stablehlo.constant dense<[[1, 2, 3]]> : tensor<1x3xi64>
+  %output_dimensions = stablehlo.constant dense<[2, 3, 2]> : tensor<3xi64>
+  %result = "stablehlo.dynamic_broadcast_in_dim"(%operand, %output_dimensions) {
+    broadcast_dimensions = array<i64: 2, 1>,
+    known_expanding_dimensions = array<i64: 0>,
+    known_non_expanding_dimensions = array<i64: 1>
```

Comment:
```  
known_expanding_dimensions = array<i64: 1>,
  known_non_expanding_dimensions = array<i64: 0>
```

Reply:
```
 known_expanding_dimensions = array<i64: 0>,
  known_non_expanding_dimensions = array<i64: 1>

```
is correct for 
```
// %operand: [
//            [1, 2, 3]
//           ] 

to 

// %result: [
//            [
//             [1, 1],
//             [2, 2],
//             [3, 3]
//            ],
//            [
//             [1, 1],
//             [2, 2],
//             [3, 3]
//            ]
//          ]
```
operand dim 0 size = 1
result dim 0 size = 2